webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/UploadService.scala

```diff
@@ -224,12 +228,50 @@ class UploadService @Inject()(dataSourceRepository: DataSourceRepository,
       Fox.successful(())
     else {
       for {
-        _ <- addLayerAndResolutionDirIfMissing(unpackToDir).toFox
-        _ <- addSymlinksToOtherDatasetLayers(unpackToDir, layersToLink.getOrElse(List.empty))
-        _ <- addLinkedLayersToDataSourceProperties(unpackToDir, dataSourceId.team, layersToLink.getOrElse(List.empty))
+        isZarr <- looksLikeZarrArray(unpackToDir).toFox
+
+        _ <- Fox.runIf(isZarr)(exploreLocalDatasource(unpackToDir, dataSourceId))
+        p <- Fox.runIf(!isZarr)(tryExploringMultipleZarrLayers(unpackToDir, dataSourceId))
+        _ <- Fox.runIf(!isZarr && p.flatten.isEmpty)(
+          postProcessUploadedWKWDataSource(unpackToDir, dataSourceId, layersToLink))
```

Comment:
It would be great if the layersToLink feature would also work with uploaded zarr datasets. This adds entries to layers from other datasets as specified by the layersToLink parameter